### PR TITLE
Added more fields to chat preview structure

### PR DESCRIPTION
**Pull request by vkjr**
ChatPreview structure was extended to pass more data to mobile needed for generating the last message preview.
[Related mobile issue](https://github.com/status-im/status-mobile/issues/15281).

### CODE REVIEW

File: protocol/messenger_chats.go [hunk 1]
```diff
@@ -76,10 +75,24 @@ func (m *Messenger) ChatsPreview() []*ChatPreview {
...
+
 			if chat.LastMessage != nil {
+
+				chatPreview.OutgoingStatus = chat.LastMessage.OutgoingStatus
+				chatPreview.ResponseTo = chat.LastMessage.ResponseTo
 				chatPreview.ContentType = chat.LastMessage.ContentType
-				if chat.LastMessage.ContentType == protobuf.ChatMessage_TEXT_PLAIN {
 
+				if chat.LastMessage.ContentType == protobuf.ChatMessage_IMAGE {
+					chatPreview.ParsedText = chat.LastMessage.ParsedText
```
Comment (cammellos) [inline protocol/messenger_chats.go:89]: `GetSimplifiedText`?

**Review by cammellos — APPROVED** [review]

**Comment by vkjr** [comment]
@cammellos, could you please take a look at this PR again?
Per design in mobile app you should show a "message deleted" preview, when the last message is a deleted one. So I changed a few more places in the code to leave a deleted message in `chat.LastMessage` field. 
The only other logic where `chat.LastMessage` was used was the one with notifications, so I updated it a bit too.

**Comment by cammellos** [comment]
@vkjr looks good, nice we can simplify, would you mind adding (or updating?) tests please?

**Comment by vkjr** [comment]
@cammellos, thanks, I've updated tests and also found a bug that wasn't obvious before.

**Review by cammellos — APPROVED** [review]

**Pull request merged**